src/main/java/org/opentripplanner/updater/transportation_network_company/lyft/LyftTransportationNetworkCompanyDataSource.java

```diff
@@ -61,51 +58,35 @@ public LyftTransportationNetworkCompanyDataSource(String baseUrl, String clientI
         this.clientSecret = clientSecret;
     }
 
-    private String getAccessToken() throws IOException {
-        // check if token needs to be obtained
-        Date now = new Date();
-        if (tokenExpirationTime == null || now.after(tokenExpirationTime)) {
-            // token needs to be obtained
-            LOG.info("Requesting new lyft access token");
-
-            ObjectMapper mapper = new ObjectMapper();
-            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-
-            // prepare request to get token
-            UriBuilder uriBuilder = UriBuilder.fromUri(baseUrl + "oauth/token");
-            URL url = new URL(uriBuilder.toString());
-            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-            String userpass = clientId + ":" + clientSecret;
-            String basicAuth = "Basic " + javax.xml.bind.DatatypeConverter.printBase64Binary(userpass.getBytes());
-            connection.setRequestProperty("Authorization", basicAuth);
-            connection.setRequestProperty("Content-Type", "application/json;charset=UTF-8");
-
-            // set request body
-            LyftAuthenticationRequestBody authRequest = new LyftAuthenticationRequestBody(
-                "client_credentials",
-                "public"
-            );
-            connection.setDoOutput(true);
-            mapper.writeValue(connection.getOutputStream(), authRequest);
-
-            // send request and parse repsonse
-            InputStream responseStream = connection.getInputStream();
-            LyftAuthenticationResponse response = mapper.readValue(responseStream, LyftAuthenticationResponse.class);
-            accessToken = response.access_token;
-            tokenExpirationTime = new Date();
-            tokenExpirationTime.setTime(tokenExpirationTime.getTime() + (response.expires_in - 60) * 1000);
-
-            LOG.info("Received new lyft access token");
-        }
-
-        return accessToken;
-    }
-
     @Override
     public TransportationNetworkCompany getTransportationNetworkCompanyType() {
         return TransportationNetworkCompany.LYFT;
     }
 
+    @Override
+    protected HttpURLConnection buildOAuthConnection() throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
The mapper can be a static field.

Reply:
Updated in https://github.com/ibi-group/OpenTripPlanner/pull/115/commits/0db7e55f333a0d43560d81900de07bcf56562db7, and for Uber too.